### PR TITLE
Create a library of constants

### PR DESCRIPTION
**Pull request by rouille**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
Create a library of constants that can be used for PyPSA importer/exporter and section of the code relaying on `Grid` templates.

### What the code is doing
It factors out constants, gather them in new modules located in **powersimdata/input/const/**, add information to the `AbstractGrid` class and refactor the `MockGrid` class 

### Testing
Existing unit tests

### Where to look
Besides factoring out some constants and relocate them in new modules, there is a refactor of the `MockGrid` object. This is done in `powersimdata.tests.mock_grid`

### Usage Example/Visuals
N/A

### Time estimate
15min

### CODE REVIEW

File: powersimdata/tests/mock_grid.py [hunk 1]
```diff
@@ -1,85 +1,23 @@
...
+class MockGrid(AbstractGrid):
     def __init__(self, grid_attrs=None, model="usa_tamu"):
         """Constructor.
 
         :param dict grid_attrs: dictionary of {*field_name*, *data_dict*} pairs
-            where *field_name* is the name of the data frame (sub, bus2sub,
-            branch, bus, dcline, gencost or plant) and *data_dict* a dictionary
-            in which the keys are the column name of the data frame associated
-            to *field_name* and the values are a list of values.
+            where *field_name* can be: sub, bus2sub, branch, bus, dcline, plant, plant, gencost_before, gencost_after, storage_gen, storage_StorageData and *data_dict* is a dictionary in which the keys are the column name of the data frame associated to *field_name* and the values are a list of values.
```
Comment (BainanXia) [inline powersimdata/tests/mock_grid.py:14]: Maybe we should break this long line into smaller ones.

Comment (rouille) [reply powersimdata/tests/mock_grid.py:14]: For sure

**Review by BainanXia — APPROVED** [review]
Looks good. Thanks!

File: powersimdata/tests/mock_grid.py (hunk 1 above)
Comment (rouille) [reply powersimdata/tests/mock_grid.py]: Done

File: powersimdata/input/converter/pypsa_to_grid.py [hunk 2]
```diff
@@ -432,7 +437,8 @@ def _translate_pnl(self, pnl, key):
         data frames.
 
         :param str pnl: name of the time-dependent dataframe.
-        :param str key: key in the :data:`pypsa_const` dictionary.
+        :param str key: key in :data:`powersimdata.input.const.pypsa_const.pypsa.const`
```
Comment (jenhagg) [inline powersimdata/input/converter/pypsa_to_grid.py:440]: Should this be `powersimdata.input.const.pypsa_const.pypsa_const`?

Comment (rouille) [reply powersimdata/input/converter/pypsa_to_grid.py]: Done

**Review by jenhagg — APPROVED** [review]

**Pull request merged**